packages/shared/lib/sdk/sync.ts

```diff
@@ -392,6 +392,7 @@ export interface NangoProps {
     runnerFlags: RunnerFlags;
     debug: boolean;
     startedAt: Date;
+    endUser: { id: number; endUserId: string | null; orgId: string | null } | null;
```

Comment:
we passing the endUser only so we can have it when handling the job result. Have you thought about fetching it from the db when starting the jobs AND again when handling the job result?

Reply:
yes, I'm not a fan. The way I see it, a job should be self-contained you send a payload and you receive the same thing, plus querying a new thing is one more possible way to fail. Tbf almost everything we put in NangoProps is not for execution per se.
Wdyt?

Reply:
I like the self-contained aspect of NangoProps but that's a lot of data doing the round trip. I am not sure what makes the most sense tbh